src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs

```diff
@@ -197,12 +198,31 @@ private static ObjectType SetBicepResourceProperties(ObjectType objectType, Reso
                 properties = properties.SetItem("subscriptionId", new TypeProperty("subscriptionId", LanguageConstants.String, TypePropertyFlags.DeployTimeConstant));
             }
 
+            var functions = GetTypeBicepFunctions(typeReference);
+
             return new ObjectType(
                 objectType.Name,
                 objectType.ValidationFlags,
                 isExistingResource ? ConvertToReadOnly(properties.Values) : properties.Values,
                 objectType.AdditionalPropertiesType,
-                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags);
+                isExistingResource ? ConvertToReadOnly(objectType.AdditionalPropertiesFlags) : objectType.AdditionalPropertiesFlags,
+                functions);
+        }
+
+        private static IEnumerable<FunctionOverload> GetTypeBicepFunctions(ResourceTypeReference resourceType)
+        {
+            switch (resourceType.FullyQualifiedType.ToLowerInvariant())
+            {
+                case "microsoft.keyvault/vaults":
+                    yield return new FunctionOverloadBuilder("getSecret")
+                        .WithReturnType(LanguageConstants.KeyVaultSecretReference)
+                        .WithDescription("References a secret from this key vault to be binded to a secure string module parameter")
```

Comment:
What about "Gets a reference to a key vault secret, which can be provided to a secure string module parameter"?

Reply:
+1